src/core/installAddon.js

```diff
@@ -0,0 +1,192 @@
+/* global CustomEvent, document, window */
+import { connect } from 'react-redux';
+import config from 'config';
+
+import log from 'core/logger';
+import tracking, { getAction } from 'core/tracking';
+import {
+  CLOSE_INFO,
+  DISABLED,
+  DOWNLOAD_FAILED,
+  DOWNLOAD_PROGRESS,
+  ENABLED,
+  ERROR,
+  FATAL_ERROR,
+  FATAL_INSTALL_ERROR,
+  FATAL_UNINSTALL_ERROR,
+  INSTALL_CATEGORY,
+  INSTALL_ERROR,
+  INSTALL_FAILED,
+  INSTALL_STATE,
+  SET_ENABLE_NOT_AVAILABLE,
+  SHOW_INFO,
+  START_DOWNLOAD,
+  UNINSTALL_CATEGORY,
+  UNINSTALLED,
+  UNINSTALLING,
+} from 'core/constants';
+import * as addonManager from 'core/addonManager';
+
+export function makeProgressHandler(dispatch, guid) {
+  return (addonInstall, e) => {
```

Comment:
I always avoided using `event` because it was highlighted differently in my edit back in the day. I finally bothered to look into it and it looks like (at least in Firefox) `event` is set if you have something like `<p onClick="doSomething(event)">` but it appears it is not set anywhere else. I'll change it.

🌟 The more you (or at least I) know.